File: NOVEMBER-2025/18-11-2025/p717.rb
# frozen_string_literal: true
# @param {Integer[]} bits
# @return {Boolean}
def is_one_bit_character(bits)
  len = bits.length
  if len == 1
    return true
  end
  if len == 2 and bits[0] == 1
    return false
  end
  cur = 0
  while cur < len
    if bits[cur] == 1
      if len - cur == 2
        return false
      end
      cur += 2
      if len - cur == 1
        return true
      end
    else
      cur += 1
    end
  end
  true
end

File: NOVEMBER-2025/18-11-2025/p717.erl
-spec is_one_bit_character(Bits :: [integer()]) -> boolean().
is_one_bit_character(Bits) ->
    is_one_bit_character(Bits, 0, length(Bits)).

is_one_bit_character(Bits, I, N) when I < N - 1 ->
    case lists:nth(I + 1, Bits) of
        1 -> is_one_bit_character(Bits, I + 2, N);
        0 -> is_one_bit_character(Bits, I + 1, N)
    end;
is_one_bit_character(_Bits, I, N) ->
    I =:= N - 1.